File: Adapter/Go/adapter/LogFileOperate.go
package adapter

import "fmt"

type LogFileOperate struct {
  LogFileOperateAPI
  file_name string
}

func NewLogFileOperate(file_name string) *LogFileOperate{
  return &LogFileOperate{file_name: file_name}
}

func (this *LogFileOperate) ReadLogFile() string {
  content := "test content from " + this.file_name
  return content
}

func (this *LogFileOperate) WriteLogFile(content string) {
  fmt.Printf("Writing %s\n", content)
}

File: Adapter/PHP/LogFileOperate.php
<?php
  class LogFileOperate implements LogFileOperateAPI {
    private $file_name;

    public function __construct($file_name) {
      $this->file_name = $file_name;
    }

    public function readLogFile() {
      $content = "test content from " . $this->file_name;
      return $content;
    }

    public function writeLogFile($content) {
      echo "writing $content\n";
    }
  }
?>
